### ISSUE TITLE
Using js.fetch with POST

### ISSUE THREAD

**Issue by dmondev** [pyodide/pyodide#1443]
Sorry if this might be trivial, but the following doesn't seem to work from pyodide:

```
  resp = await js.fetch('/someurl', {
      "method": "POST"
    , "body": '{ "some" : "json" }'
    , "credentials": "same-origin"
    , "headers": { "Content-Type": "application/json" }    
  })
```

The second param is ignored, and this generates a GET request.

**Comment by dmondev** [comment]
Sorry, I missed this was already covered. should I close this as duplicate ?

**Comment by rth** [comment]
No worries. Yes, let's close it, but don't hesitate to comment in the future if it's still an issue.

**Comment by hoodmane** [comment]
@dmondev I think this deserves a little extra explanation -- we should at least add an FAQ example or something. There are a couple of ways to do this:

Since #1433 was merged, it's possible to use key word arguments:
```py
resp = await fetch('/someurl',
    method= "POST", 
    body= '{ "some" : "json" }',
    credentials= "same-origin",
    headers={ "Content-Type": "application/json" },
)
```
however this still won't work because Javascript won't recognize what to do with the headers argument. The easiest way to do this without a memory leak is to use the `to_js` function created on branch #1436:
```py
from pyodide import to_js
fetch(...,
    headers=to_js({ "Content-Type": "application/json" }),
)
```
Without `to_js`, one method I've noticed without memory leaks is:
```py
from js import Object
resp = await fetch('/someurl',
    ...,
    headers=Object.new(**{ "Content-Type": "application/json" }),
)
```
If you are okay leaking the `dict` then the following works also:
```
    # leak the dict!
    headers={ "Content-Type": "application/json" }.items(),
```

**Comment by hoodmane** [comment]
Also, with to_js:
```
  from pyodide import to_js
  resp = await js.fetch('/someurl', to_js({
      "method": "POST"
    , "body": '{ "some" : "json" }'
    , "credentials": "same-origin"
    , "headers": { "Content-Type": "application/json" }    
  }))
```

**Comment by hoodmane** [comment]
I'm going to reopen this until we add more documentation about this.

**Comment by hoodmane** [comment]
Closed by #1484.
